src/Obsidian/TasksEvents.ts

```diff
@@ -69,6 +70,20 @@ export class TasksEvents {
         this.obsidianEvents.trigger(Event.ReloadOpenSearchResults);
     }
 
+    // ------------------------------------------------------------------------
+    // ReloadVault event
+
+    public onReloadVault(handler: () => void): EventRef {
+        this.logger.debug('TasksEvents.onReloadVault()');
+        const name = Event.ReloadVault;
+        return this.obsidianEvents.on(name, handler);
+    }
+
+    public triggerReloadVault(): void {
+        this.logger.debug('TasksEvents.triggerReloadVault()');
+        this.obsidianEvents.trigger(Event.ReloadVault);
+    }
+
```

Comment:
I need to have a think about this.

We added it when we thought we could implement the change "properly" via events, but then we couldn't get the event code working - so we fell back on a direct all inside `SettingsTab.ts`, temporarily.

Reply:
Sorry - I was overcome with the urge to have another go at taking an events-based approach, instead of waiting until our next session!